### PR TITLE
Release/1.2.2

### LINKED ISSUE #312: java.lang.NullPointerException: Attempt to invoke virtual method org.fundacionparaguaya.advisorapp.models.ResponseType org.fundacionparaguaya.advisorapp.models.BackgroundQuestion.getResponseType() on a null object reference
**Number:** 	 148

**Status:** 	 New

**Reported At:** 	 2018-02-21 19:14:49 UTC

**Email:** 	 

**URL:** 	 https://insta.co/crashes/E8KBLod2

**App Version:** 	 v1.2.1 (Baron) (6)

**Current View:** 	 org.fundacionparaguaya.advisorapp.activities.SurveyActivity

**Device:** 	 samsung SM-T285

**Location:** 	 Avellaneda, Argentina

**Duration:** 	 0

**Connectivity:** 	 Carrier: , WiFi: krypton

**Memory:** 	 753.0/1416.0 MB

**Storage:** 	 3.035/4.897 GB

**Orientation:** 	 landscape

**Screen Size:** 	 1280x800

**Density:** 	 mdpi

**Battery:** 	 63, Unplugged

**User Data:** 	 

**User Steps:**
```
1519239926679 org.fundacionparaguaya.advisorapp.activities.LoginActivity was created.

1519239926736 org.fundacionparaguaya.advisorapp.activities.LoginActivity was started.

1519239926900 org.fundacionparaguaya.advisorapp.activities.LoginActivity was resumed.

1519240188412 org.fundacionparaguaya.advisorapp.activities.LoginActivity was paused.

1519240188437 org.fundacionparaguaya.advisorapp.activities.DashActivity was created.

1519240188532 org.fundacionparaguaya.advisorapp.activities.DashActivity was started.

1519240188557 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519240198433 org.fundacionparaguaya.advisorapp.activities.LoginActivity was stopped.

1519240198435 org.fundacionparaguaya.advisorapp.activities.LoginActivity was destroyed.

1519240238780 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519240238798 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519240238838 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.

1519240238839 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was resumed.

1519240239906 org.fundacionparaguaya.advisorapp.activities.DashActivity was stopped.

1519240488912 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was paused.

1519240488934 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was stopped.

1519240488968 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was destroyed.

1519240489025 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519240489057 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.


```

**Console Log:**
```

1519240468665  D/TextView( 8533): setTypeface with style : 0

1519240468665  D/TextView( 8533): setTypeface with style : 0

1519240468675  D/TextView( 8533): setTypeface with style : 0

1519240468675  D/TextView( 8533): setTypeface with style : 0

1519240469136  I/art     ( 8533): Background partial concurrent mark sweep GC freed 191510(9MB) AllocSpace objects, 33(1419KB) LOS objects, 39 free, 21MB/36MB, paused 1.312ms total 123.077ms

1519240469256  D/ViewRootImpl( 8533): ViewPostImeInputStage ACTION_DOWN

1519240480227  I/art     ( 8533): Background partial concurrent mark sweep GC freed 213730(10MB) AllocSpace objects, 27(1312KB) LOS objects, 39 free, 22MB/37MB, paused 1.434ms total 110.137ms

1519240487644  D/ViewRootImpl( 8533): ViewPostImeInputStage ACTION_DOWN

1519240488615  D/ViewRootImpl( 8533): ViewPostImeInputStage ACTION_DOWN

1519240488705  D/TextView( 8533): setTypeface with style : 0

1519240488705  D/TextView( 8533): setTypeface with style : 0

1519240488915  D/SensorManager( 8533): unregisterListener ::   

1519240488925  D/TextView( 8533): setTypeface with style : 0

1519240488925  D/TextView( 8533): setTypeface with style : 0

1519240488965  D/InputMethodManager( 8533): windowDismissed mLockisused = false

1519240488975  W/ContextImpl( 8533): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240488985  W/ContextImpl( 8533): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240489035  D/PhoneWindow( 8533): *FMB* installDecor mIsFloating : false

1519240489035  D/PhoneWindow( 8533): *FMB* installDecor flags : -2139029248

1519240489035  D/TextView( 8533): setTypeface with style : 0

1519240489035  D/TextView( 8533): setTypeface with style : 0

1519240489045  D/TextView( 8533): setTypeface with style : 0

1519240489055  I/System.out( 8533): (HTTPLog)-Static: isSBSettingEnabled false

1519240489065  I/System.out( 8533): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519240489075  D/TextView( 8533): setTypeface with style : 0

1519240489075  D/TextView( 8533): setTypeface with style : 0

1519240489085  D/AndroidRuntime( 8533): Shutting down VM

1519240489085  W/GooglePlayServicesUtil( 8533): Google Play services out of date.  Requires 11910000 but found 11302230

1519240489125  W/ContextImpl( 8533): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240489125  W/ContextImpl( 8533): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240489145  I/System.out( 8533): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519240489185  I/System.out( 8533): (HTTPLog)-Static: isSBSettingEnabled false

1519240489185  I/System.out( 8533): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519240489195  I/System.out( 8533): KnoxVpnUidStorageknoxVpnSupported API value returned is false


```

**Instabug Log:**
```

```

### LINKED ISSUE #313: java.lang.NullPointerException: Attempt to invoke virtual method java.lang.String org.fundacionparaguaya.advisorapp.models.BackgroundQuestion.getDescription() on a null object reference
**Number:** 	 149

**Status:** 	 New

**Reported At:** 	 2018-02-21 19:16:05 UTC

**Email:** 	 

**URL:** 	 https://insta.co/crashes/w5OigUyu

**App Version:** 	 v1.2.1 (Baron) (6)

**Current View:** 	 org.fundacionparaguaya.advisorapp.activities.SurveyActivity

**Device:** 	 samsung SM-T285

**Location:** 	 Avellaneda, Argentina

**Duration:** 	 0

**Connectivity:** 	 Carrier: , WiFi: krypton

**Memory:** 	 746.0/1416.0 MB

**Storage:** 	 3.036/4.897 GB

**Orientation:** 	 landscape

**Screen Size:** 	 1280x800

**Density:** 	 mdpi

**Battery:** 	 63, Unplugged

**User Data:** 	 

**User Steps:**
```
1519240498513 org.fundacionparaguaya.advisorapp.activities.DashActivity was created.

1519240498686 org.fundacionparaguaya.advisorapp.activities.DashActivity was started.

1519240498733 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519240511782 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519240511800 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519240511827 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.

1519240511828 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was resumed.

1519240512897 org.fundacionparaguaya.advisorapp.activities.DashActivity was stopped.

1519240565248 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was paused.

1519240565268 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was stopped.

1519240565299 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was destroyed.

1519240565344 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519240565374 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.


```

**Console Log:**
```
240561856  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561866  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561876  D/TextView( 9002): setTypeface with style : 0

1519240561886  D/TextView( 9002): setTypeface with style : 0

1519240561936  I/System.out( 9002): (HTTPLog)-Static: isSBSettingEnabled false

1519240561936  I/System.out( 9002): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519240561946  I/System.out( 9002): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519240562587  D/ViewRootImpl( 9002): ViewPostImeInputStage ACTION_DOWN

1519240563488  D/ViewRootImpl( 9002): ViewPostImeInputStage ACTION_DOWN

1519240564459  D/ViewRootImpl( 9002): ViewPostImeInputStage ACTION_DOWN

1519240565019  D/ViewRootImpl( 9002): ViewPostImeInputStage ACTION_DOWN

1519240565079  D/TextView( 9002): setTypeface with style : 0

1519240565079  D/TextView( 9002): setTypeface with style : 0

1519240565240  D/SensorManager( 9002): unregisterListener ::   

1519240565250  D/TextView( 9002): setTypeface with style : 0

1519240565250  D/TextView( 9002): setTypeface with style : 0

1519240565290  D/InputMethodManager( 9002): windowDismissed mLockisused = false

1519240565300  W/ContextImpl( 9002): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240565300  W/ContextImpl( 9002): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240565340  D/PhoneWindow( 9002): *FMB* installDecor mIsFloating : false

1519240565340  D/PhoneWindow( 9002): *FMB* installDecor flags : -2139029248

1519240565350  D/TextView( 9002): setTypeface with style : 0

1519240565350  D/TextView( 9002): setTypeface with style : 0

1519240565350  D/TextView( 9002): setTypeface with style : 0

1519240565370  I/System.out( 9002): (HTTPLog)-Static: isSBSettingEnabled false

1519240565370  I/System.out( 9002): (HTTPLog)-Static: isSBSettingEnabled false

1519240565390  D/TextView( 9002): setTypeface with style : 0

1519240565390  D/AndroidRuntime( 9002): Shutting down VM

1519240565420  W/ContextImpl( 9002): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519240565430  W/ContextImpl( 9002): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache


```

**Instabug Log:**
```

```

### LINKED ISSUE #314: java.lang.NullPointerException: Attempt to invoke virtual method android.content.Context android.view.ViewGroup.getContext() on a null object reference
**Number:** 	 150

**Status:** 	 New

**Reported At:** 	 2018-02-21 19:42:52 UTC

**Email:** 	 

**URL:** 	 https://insta.co/crashes/OhpAkqwu

**App Version:** 	 v1.2.2 (Roadrunner) (7)

**Current View:** 	 org.fundacionparaguaya.advisorapp.activities.SurveyActivity

**Device:** 	 samsung SM-T285

**Location:** 	 Avellaneda, Argentina

**Duration:** 	 0

**Connectivity:** 	 Carrier: , WiFi: krypton

**Memory:** 	 769.0/1416.0 MB

**Storage:** 	 3.047/4.897 GB

**Orientation:** 	 landscape

**Screen Size:** 	 1280x800

**Density:** 	 mdpi

**Battery:** 	 58, Unplugged

**User Data:** 	 

**User Steps:**
```
1519241961911 org.fundacionparaguaya.advisorapp.activities.LoginActivity was created.

1519241961975 org.fundacionparaguaya.advisorapp.activities.LoginActivity was started.

1519241962133 org.fundacionparaguaya.advisorapp.activities.LoginActivity was resumed.

1519241978876 org.fundacionparaguaya.advisorapp.activities.LoginActivity was paused.

1519241978907 org.fundacionparaguaya.advisorapp.activities.DashActivity was created.

1519241978996 org.fundacionparaguaya.advisorapp.activities.DashActivity was started.

1519241979024 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519241988897 org.fundacionparaguaya.advisorapp.activities.LoginActivity was stopped.

1519241988898 org.fundacionparaguaya.advisorapp.activities.LoginActivity was destroyed.

1519241992177 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519241995021 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519241996268 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519241998366 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519242039886 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519242039904 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519242039935 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.

1519242039936 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was resumed.

1519242041010 org.fundacionparaguaya.advisorapp.activities.DashActivity was stopped.

1519242042721 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was paused.

1519242042737 org.fundacionparaguaya.advisorapp.activities.DashActivity was started.

1519242042738 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519242052740 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was stopped.

1519242052741 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was destroyed.

1519242054758 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519242054781 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519242054808 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.

1519242054808 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was resumed.

1519242055863 org.fundacionparaguaya.advisorapp.activities.DashActivity was stopped.

1519242172290 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was paused.

1519242172326 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was stopped.

1519242172367 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was destroyed.

1519242172418 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519242172446 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.


```

**Console Log:**
```
(11381): Background partial concurrent mark sweep GC freed 211631(10MB) AllocSpace objects, 25(1454KB) LOS objects, 39 free, 21MB/36MB, paused 1.251ms total 110.382ms

1519242164400  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242165190  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242166972  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242167893  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242168273  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242168934  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242169795  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242170736  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242171266  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242172007  D/ViewRootImpl(11381): ViewPostImeInputStage ACTION_DOWN

1519242172087  D/TextView(11381): setTypeface with style : 0

1519242172087  D/TextView(11381): setTypeface with style : 0

1519242172277  D/SensorManager(11381): unregisterListener ::   

1519242172297  D/TextView(11381): setTypeface with style : 0

1519242172297  D/TextView(11381): setTypeface with style : 0

1519242172357  D/InputMethodManager(11381): windowDismissed mLockisused = false

1519242172357  W/ContextImpl(11381): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519242172367  W/ContextImpl(11381): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519242172418  D/PhoneWindow(11381): *FMB* installDecor mIsFloating : false

1519242172418  D/PhoneWindow(11381): *FMB* installDecor flags : -2139029248

1519242172418  D/TextView(11381): setTypeface with style : 0

1519242172418  D/TextView(11381): setTypeface with style : 0

1519242172428  D/TextView(11381): setTypeface with style : 0

1519242172438  I/System.out(11381): (HTTPLog)-Static: isSBSettingEnabled false

1519242172448  I/System.out(11381): (HTTPLog)-Static: isSBSettingEnabled false

1519242172448  I/System.out(11381): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519242172448  D/AndroidRuntime(11381): Shutting down VM

1519242172468  W/GooglePlayServicesUtil(11381): Google Play services out of date.  Requires 11910000 but found 11302230

1519242172488  W/ContextImpl(11381): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519242172498  W/ContextImpl(11381): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519242172508  I/System.out(11381): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519242172588  I/System.out(11381): (HTTPLog)-Static: isSBSettingEnabled false

1519242172588  I/System.out(11381): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519242172598  I/System.out(11381): KnoxVpnUidStorageknoxVpnSupported API value returned is false


```

**Instabug Log:**
```

```

### LINKED ISSUE #315: java.lang.NullPointerException: Attempt to invoke interface method void org.fundacionparaguaya.advisorapp.fragments.callbacks.BackgroundQuestionCallback.onQuestionAnswered(org.fundacionparaguaya.advisorapp.models.BackgroundQuestion, java.lang.Object) on a
**Number:** 	 151

**Status:** 	 New

**Reported At:** 	 2018-02-21 19:57:26 UTC

**Email:** 	 

**URL:** 	 https://insta.co/crashes/Yh_its2X

**App Version:** 	 v1.2.2 (Roadrunner) (5)

**Current View:** 	 org.fundacionparaguaya.advisorapp.activities.SurveyActivity

**Device:** 	 samsung SM-T285

**Location:** 	 Avellaneda, Argentina

**Duration:** 	 0

**Connectivity:** 	 Carrier: , WiFi: krypton

**Memory:** 	 766.0/1416.0 MB

**Storage:** 	 3.046/4.897 GB

**Orientation:** 	 landscape

**Screen Size:** 	 1280x800

**Density:** 	 mdpi

**Battery:** 	 58, Unplugged

**User Data:** 	 

**User Steps:**
```
1519242734962 org.fundacionparaguaya.advisorapp.activities.LoginActivity was created.

1519242735026 org.fundacionparaguaya.advisorapp.activities.LoginActivity was started.

1519242735205 org.fundacionparaguaya.advisorapp.activities.LoginActivity was resumed.

1519242754769 org.fundacionparaguaya.advisorapp.activities.LoginActivity was paused.

1519242754801 org.fundacionparaguaya.advisorapp.activities.DashActivity was created.

1519242754893 org.fundacionparaguaya.advisorapp.activities.DashActivity was started.

1519242754923 org.fundacionparaguaya.advisorapp.activities.DashActivity was resumed.

1519242764792 org.fundacionparaguaya.advisorapp.activities.LoginActivity was stopped.

1519242764793 org.fundacionparaguaya.advisorapp.activities.LoginActivity was destroyed.

1519242780326 org.fundacionparaguaya.advisorapp.activities.DashActivity was paused.

1519242780344 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519242780372 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.

1519242780373 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was resumed.

1519242781465 org.fundacionparaguaya.advisorapp.activities.DashActivity was stopped.

1519243045542 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was paused.

1519243045567 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was stopped.

1519243045607 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was destroyed.

1519243045651 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was created.

1519243045682 org.fundacionparaguaya.advisorapp.activities.SurveyActivity was started.


```

**Console Log:**
```
 with style : 0

1519243043858  D/TextView(12587): setTypeface with style : 0

1519243043858  D/TextView(12587): setTypeface with style : 0

1519243043858  D/TextView(12587): setTypeface with style : 0

1519243043858  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243043868  D/TextView(12587): setTypeface with style : 0

1519243044588  D/ViewRootImpl(12587): ViewPostImeInputStage ACTION_DOWN

1519243045329  D/ViewRootImpl(12587): ViewPostImeInputStage ACTION_DOWN

1519243045369  D/TextView(12587): setTypeface with style : 0

1519243045369  D/TextView(12587): setTypeface with style : 0

1519243045529  D/SensorManager(12587): unregisterListener ::   

1519243045549  D/TextView(12587): setTypeface with style : 0

1519243045549  D/TextView(12587): setTypeface with style : 0

1519243045599  D/InputMethodManager(12587): windowDismissed mLockisused = false

1519243045599  W/ContextImpl(12587): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519243045609  W/ContextImpl(12587): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519243045649  D/PhoneWindow(12587): *FMB* installDecor mIsFloating : false

1519243045649  D/PhoneWindow(12587): *FMB* installDecor flags : -2139029248

1519243045659  D/TextView(12587): setTypeface with style : 0

1519243045659  D/TextView(12587): setTypeface with style : 0

1519243045659  D/TextView(12587): setTypeface with style : 0

1519243045680  I/System.out(12587): (HTTPLog)-Static: isSBSettingEnabled false

1519243045680  I/System.out(12587): (HTTPLog)-Static: isSBSettingEnabled false

1519243045680  I/System.out(12587): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519243045690  D/TextView(12587): setTypeface with style : 0

1519243045700  D/TextView(12587): setTypeface with style : 0

1519243045700  D/AndroidRuntime(12587): Shutting down VM

1519243045710  W/GooglePlayServicesUtil(12587): Google Play services out of date.  Requires 11910000 but found 11302230

1519243045720  W/ContextImpl(12587): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519243045740  W/ContextImpl(12587): Failed to ensure directory: /storage/extSdCard/Android/data/org.fundacionparaguaya.advisorapp/cache

1519243045750  W/ProcessManager(12587): waitpid on failed exec failed: No child processes

1519243045790  I/System.out(12587): (HTTPLog)-Static: isSBSettingEnabled false

1519243045800  I/System.out(12587): KnoxVpnUidStorageknoxVpnSupported API value returned is false

1519243045850  I/System.out(12587): KnoxVpnUidStorageknoxVpnSupported API value returned is false


```

**Instabug Log:**
```

```

### PR DESCRIPTION
**Pull request by bhylak**
Rewrote QuestionFragment to handle rotation bug. 

Fixes #315, fixes #312, fixes #313, fixes #314

### CODE REVIEW

**Comment by bhylak** [comment]
@krconv can you review this? need to get this merged in.

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyEconomicQuestionsFragment.java [hunk 1]
```diff
@@ -38,20 +42,48 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+
+        mSharedSurveyViewModel.getEconomicResponses().observe(this, (responses)->
+        {
+            checkConditions();
+        });
 
         super.initQuestionList();
     }
 
+    //*********************For survey questions
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyEconomicQuestionsFragment.java]: should be a `region` tag

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyEconomicQuestionsFragment.java [hunk 2]
```diff
@@ -38,20 +42,48 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+        return mSharedSurveyViewModel.getBackgroundResponse(question);
+    }
+
+    @Override
+    public void onResponse(BackgroundQuestion question, String s) {
+        mSharedSurveyViewModel.setBackgroundResponse(question, s);
+        checkConditions();
+    }
+
+    //*********************For review page
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyEconomicQuestionsFragment.java]: Should be a `region` tag

File: app/src/main/java/org/fundacionparaguaya/advisorapp/activities/SurveyActivity.java [hunk 3]
```diff
@@ -122,6 +112,7 @@ protected void onCreate(@Nullable Bundle savedInstanceState)
             mIs7Inch = false;
         }
 
+        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/activities/SurveyActivity.java]: Currently this is locked on one orientation (doesn't rotate when device is changed 180 degrees); maybe [`ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE`](https://developer.android.com/reference/android/content/pm/ActivityInfo.html#SCREEN_ORIENTATION_SENSOR_LANDSCAPE) would be better?

Comment (bhylak) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/activities/SurveyActivity.java]: good call.

**Pull request merged**